### ISSUE TITLE
Tabs defined in AppShell.xaml does not invoke its view model's IQueryAttributable.ApplyQueryAttributes(query) implementaion

### ISSUE THREAD

**Issue by dk-mushiyoke** [dotnet/maui#13537]
### Description

When doing shell navigation with `Shell.Current.GoToAsync(state)` in a maui project and the destination page's view model implements `IQueryAttributable`, the implementation of `IQueryAttributable.ApplyQueryAttributes(query)` will be called to pass on query attributes, even when no data is passed into `Shell.Current.GoToAsync(state)`.

However, this behavior is observed to be missing from Tabs defined in `AppShell.xaml`, with view models that also implement `IQueryAttributable` but their implementation of `IQueryAttributable.ApplyQueryAttributes(query)` were not called.

### Steps to Reproduce

For tabs:
1. Create a new MAUI app
2. Define a tab bar in AppShell.xaml with some new `Page`s as its tabs
3. Add view model classes for each of the tabbed pages that would implement `IQueryAttributable`
4. Put breakpoints or console logs in vm's `ApplyQueryAttributes(query)` method
5. Run the app and switch between the tabs

Expected behavior: debugger breaks at `ApplyQueryAttributes(query)`/console log is displayed
Actual behavior: no breakpoints hit/no console logs displayed

For regular page navigation:
6. Add a new page and register its route with `Routing.RegisterRoute(nameof(NewPage), typeof(NewPage));`
7. Add a button to one of the pages with its `Clicked` event handler executing `Shell.Current.GoToAsync(nameof(NewPage))`
8. Repeat step 3 and 4 for the new page
9. Run the app and click on the button

Expected behavior: debugger breaks at `ApplyQueryAttributes(query)`/console log is displayed even though no query attributes are supplied
Actual behavior: matches expectation

### Link to public reproduction project repository

https://github.com/dk-mushiyoke/MauiTabQueryRepro

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android

### Affected platform versions

iOS 16.2, android 31

### Did you find any workaround?

Not yet

**Comment by dk-mushiyoke** [comment]
@PureWeen for normal page navigation `ApplyQueryAttributes` is called every time the page is navigated into as well as backed into (as in what you have described, another page performs a back navigation to this page), I think it would make sense that the tab behaviors similarly, calling `ApplyQueryAttributes` every time it is navigated to and from. Thank you for your time and attention.

**Comment by PureWeen** [comment]
Alright I see the problem @dk-mushiyoke 

The confusion here is that `QueryAttributes` aren't really applied consistently. When you navigate via `FlyoutItems` or the backbutton those will invoke `ApplyQueryAttributes` whereas if you navigate tabs, it doesn't.

If I follow you correctly then I think this request is actually at odds with https://github.com/dotnet/maui/issues/10294

So, I'll need to think on a path to make everyone happy :-)

**Comment by bakerhillpins** [comment]
> When you navigate via `FlyoutItems` or the `BackButton` those will invoke `ApplyQueryAttributes` whereas if you navigate tabs, it doesn't.
> 
> I think this request is a little bit at odds with #10294

@PureWeen I'm not sure I understand why this would be at odds with the a fore mentioned? If I'm reading this issue correctly it's saying that `ApplyQueryAttributes` is just not being called when navigating between tabs. 

Whereas the other 10294 concerns capturing objects via the `IQueryAttributable` interface so they are presented every time you navigate to a particular page.  Can you help me understand what you're seeing?

**Comment by shmoe01** [comment]
Any more thought on this one? @PureWeen was you able to figure out a way forward on this one?

**Comment by homeyf** [comment]
Verified this issue with Visual Studio Enterprise 17.8.0 Preview 1.0. Can repro on android platform with sample Project.
https://github.com/dk-mushiyoke/MauiTabQueryRepro
![image](https://github.com/dotnet/maui/assets/82450723/9feba9c9-07f6-4441-9f07-6b7f4a7b069e)

**Comment by cytoph** [comment]
The problem still exists. Reproduced with Visual Studio Enterprise 17.8.0.

**Comment by tele-bird** [comment]
> Alright I see the problem @dk-mushiyoke
> 
> When you navigate via `FlyoutItems` or the `BackButton` those will invoke `ApplyQueryAttributes` whereas if you navigate tabs, it doesn't.
> 
> I think this request is a little bit at odds with #10294
> 
> So, I'll need to think on a path to make everyone happy :-)

@PureWeen, yes, this is exactly what I'm seeing as well:  `ApplyQueryAttributes` isn't invoked when you tap through the tabs.  I think that the cause is that when you navigate by tapping tabs, Shell uses the ContentTemplate to figure out what page to navigate to, but it doesn't allow for a way to pass parameters.  So, from my perspective, you could also solve this by making ShellContent.Route bindable.  HTH

**Comment by kubaflo** [comment]
Closing as fixed. If anyone can still reproduce this on a current .NET MAUI build, please open a new issue with a fresh repro and link back here.

**Comment by kubaflo** [comment]
Reopening — this issue should not have been closed.

It carries `s/verified` (a maintainer or partner reproduced it), `s/triaged` (it already went through triage), `partner/cat 😻`, `partner/syncfusion` (a partner team is tracking it). Those labels mean the report was already confirmed as a real product defect, so it is not eligible for backlog cleanup closure. Closure here requires specific fix evidence (a merged PR that closes it, or a `fixed-in-*` label), and there was none.

Sorry for the noise, and thanks to everyone who flagged it.

> Disclosure: this comment was generated by AI.
